egs/wsj/s5/steps/tfrnnlm/lstm_fast.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+# Copyright (C) 2017 Intellisist, Inc. (Author: Hainan Xu)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# this script trains a vanilla RNNLM with TensorFlow. 
+# to call the script, do
+# python steps/tfrnnlm/lstm_fast.py --data-path=$datadir \
+#        --save-path=$savepath --vocab-path=$rnn.wordlist [--hidden-size=$size]
+#
+# One example recipe is at egs/ami/s5/local/tfrnnlm/run_vanilla_rnnlm.sh
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import sys
+
+import inspect
+import time
+
+import numpy as np
+import tensorflow as tf
+
+import reader
+
+flags = tf.flags
+logging = tf.logging
+
+flags.DEFINE_integer("hidden-size", 200, "hidden dim of RNN")
+
+flags.DEFINE_string("data-path", None,
+                    "Where the training/test data is stored.")
+flags.DEFINE_string("vocab-path", None,
+                    "Where the wordlist file is stored.")
+flags.DEFINE_string("save-path", None,
+                    "Model output directory.")
+flags.DEFINE_bool("use-fp16", False,
```

Comment:
I haven't yet. It was there because the original TF script has it and I thought it might be useful.
Do you want me to delete this? Or train another system with fp16 and report its results/speed?